stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -2226,6 +2226,18 @@ class LowerType
   }
 
   const TypeInfo *visitAnyNominalTypeRef(const TypeRef *TR) {
+    if (ExternalTypeInfo) {
+      // Ask the ExternalTypeInfo. It may be a Clang-imported type.
+      std::string MangledName;
+      if (auto N = dyn_cast<NominalTypeRef>(TR))
+        MangledName = N->getMangledName();
+      else if (auto BG = dyn_cast<BoundGenericTypeRef>(TR))
+        MangledName = BG->getMangledName();
+      if (!MangledName.empty())
+        if (auto *imported = ExternalTypeInfo->getTypeInfo(MangledName))
+          return imported;
+    }
+
     auto FD = TC.getBuilder().getFieldTypeInfo(TR);
     if (FD == nullptr || FD->isStruct()) {
```

Comment:
I wonder if it were safer to just move it into the `if (FD == nullptr || FD->isStruct())` condition?

Reply:
Would we want `ExternalTypeInfo` to be consulted when `FD->isStruct()` is true? In other words, should external lookup happen only when `FD == nullptr`

Reply:
I think there are two situations where we want to ask the external provider:
- We don't have any metadata for the type, ask the external provider.
- We have metadata for the type and it's a builtin (builtins are structs), ask the external provider as it may have more accurate information.

Reply:
in those two situations, should they be checked after `TC.getBuilder().getFieldTypeInfo(TR)`?

Reply:
currently this change checks before finding out whether either of those two cases apply.

Reply:
Yup! Just updated the patch :)